fhir-persistence-cassandra/src/main/java/com/ibm/fhir/persistence/cassandra/payload/CqlReadResource.java

```diff
@@ -52,25 +48,34 @@
 
     // The version for the resource
     private final int version;
+    
+    // The unique key used to always correctly tie this record to the RDBMS
+    private final String resourcePayloadKey;
 
+    // Elements for subsetting the resource during parse
     private final List<String> elements;
+    
+    // Do we need to uncompress the payload we've read
+    private final boolean uncompress;
 
     /**
      * Public constructor
      * @param partitionId
      * @param resourceTypeId
      * @param logicalId
+     * @param resourcePayloadKey
      * @param version
      * @param elements
      */
-    public CqlReadResource(String partitionId, int resourceTypeId, String logicalId, int version, List<String> elements) {
-        CqlDataUtil.safeId(partitionId);
-        CqlDataUtil.safeId(logicalId);
+    public CqlReadResource(String partitionId, int resourceTypeId, String logicalId, int version, String resourcePayloadKey, List<String> elements, boolean uncompress) {
```

Comment:
elsewhere we use `compress`... not sure if that will be confusing or not but wanted to note it in case we're forgetting to negate that anywhere

Reply:
I'll change it to payloadCompressed